src/pymanifold.py

```diff
@@ -125,6 +125,54 @@ def channel(self,
             self.dg.edges[port_from, port_to][key] = attr
         return
 
+    def retrieve(port_in, port_out, attr):
+      if(port_out == "None"):
+        return self.dg.edges[port_in][attr]
+      else:
+        return self.dg[port_in][port_out][attr]
```

Comment:
This is correct, however I think for ease of understanding for anyone else trying to read this code this should be written as ```return self.dg.edges[(port_in, port_out)][attr]``` so that people know that this is getting an attribute for an edge, in its current form that is a bit unintuitive to someone who doesn't know NetworkX syntax.